Renaming `base cluster` to `cluster template` (#383)
arlonproj/arlon: docs/clustertemplate.md

# Next-gen Cluster Provisioning using Cluster templates

> This proposal describes a new way of provisioning workload clusters in Arlon.

Reviewer: ```suggestion
This document describes a new way of provisioning workload clusters in Arlon.
```
This is no longer a proposal

The most significant change is the *Cluster Template* construct, which replaces the current ClusterSpec.

> To distinguish them from current generation, the ones deployed from a cluster template are called next-gen clusters or gen2 clusters.

Reviewer: ```suggestion
To distinguish them from the older gen1 clusters, the ones deployed from a cluster template are called next-gen clusters or gen2 clusters.
```

> # Next-gen Cluster Provisioning using cluster template

Reviewer 2: Captalize and add 's': **Cluster Templates**

This proposal describes a new way of provisioning workload clusters in Arlon.
The most significant change is the *Cluster Template* construct, which replaces the current ClusterSpec.
To distinguish them from current generation

> clusters, the ones deployed from a cluster template are called next-gen clusters.

Reviewer 2: add "or gen2 clusters"

### Composition

A workload cluster is composed of 2 to 3 ArgoCD application resources, which are named
based on the name of the cluster template and the workload cluster. For illustration purposes,
the following discussion assumes that the cluster template is named `capi-quickstart`, the
workload cluster is named `cluster-a`, and the optional profile is named `xxx`.

#### Cluster app

The `cluster-a` application is the *cluster app* for the cluster.
It is responsible for deploying the cluster template resources, meaning the Cluster API manifests.
It is named directly from the workload cluster name.

The application's spec uses a ApplicationSourceKustomize that points to the bcluster template's git
directory. The spec ensures that all deployed resources are configured to:

- Reside in the `cluster-a` namespace, which is deployed by the *arlon app* (see below).
This achieved by setting `app.Spec.Destination.Namespace` to the workload cluster's name
  (*this only works if the resources do not specify an explicit namespace; this requirement is

> taken care of by the "prep" step on the cluster template*).

Reviewer: ```suggestion
taken care of by the "prep" step on the *cluster template*).
```
